File: src/taburet/transactions/month_report/map.js
function(doc) {
	if (doc.doc_type && doc.doc_type == 'Transaction') {
		var year = doc.date[0]
		var month = doc.date[1]
		
		if ( year < %(year)d || ( year == %(year)d && month < %(month)d ) ) {
			for each(acc in doc.from_acc) {
				emit(acc, {before: -doc.amount, kredit:0, debet:0})
			}
			for each(acc in doc.to_acc) {
				emit(acc, {before: doc.amount, kredit:0, debet:0})
			}
		} else if ( year == %(year)d && month == %(month)d ) {
			for each(acc in doc.from_acc) {
				emit(acc, {before:0, kredit:doc.amount, debet:0})
			}
			for each(acc in doc.to_acc) {
				emit(acc, {before:0, kredit:0, debet:doc.amount})
			}
		}
	}
}

File: src/taburet/transactions/month_report/map.erl
fun({Doc}) ->
	case proplists:get_value(<<"doc_type">>, Doc) of
		<<"Transaction">> ->
			Amount = proplists:get_value(<<"amount">>, Doc),
			[Year, Month | _] = proplists:get_value(<<"date">>, Doc), 
			FromAccs = proplists:get_value(<<"from_acc">>, Doc),
			ToAccs = proplists:get_value(<<"to_acc">>, Doc),
			
			if Year < %(year)d orelse (Year == %(year)d andalso Month < %(month)d) ->
				lists:foreach(fun(AccId)-> Emit(AccId, {[{debet, 0.0}, {kredit, 0.0}, {before, -Amount}]}) end, FromAccs),
				lists:foreach(fun(AccId)-> Emit(AccId, {[{debet, 0.0}, {kredit, 0.0}, {before, Amount}]}) end, ToAccs);
			Year == %(year)d andalso Month == %(month)d ->
				lists:foreach(fun(AccId)-> Emit(AccId, {[{debet, 0.0}, {kredit, Amount}, {before, 0.0}]}) end, FromAccs),
				lists:foreach(fun(AccId)-> Emit(AccId, {[{debet, Amount}, {kredit, 0.0}, {before, 0.0}]}) end, ToAccs);
			true ->
				ok
			end;
		_ -> ok
	end
end.
